### PR TITLE
Fix PyPy compatibility in adbapi init

### PR DESCRIPTION
**Pull request by Valodim**
Following up to #2760 (and with #3098 fixed), this PR adds the simple required fix for the database initialization with adbapi to be able to run on pypy again.

I also added pypy35 to the tox and travis configs - let's see if that works :)

Signed-Off-By: Vincent Breitmoser <look@my.amazin.horse>

### CODE REVIEW

**Comment by Valodim** [comment]
[Looks like](https://travis-ci.org/matrix-org/synapse/builds/453939851) the pypy35 test went through equally well as the python35 one.

But pypy35-postgres caused a bunch of errors, mostly "too many connections" on the psql server? The pypy tests also ran significantly longer than the py3 ones, not sure what to make of that :) On the plus side though, setting up tox for pypy was straightforward and worked almost right out of the box, yay

**Comment by luminoso** [comment]
> pypy tests also ran significantly longer than the py3 ones
I think this is expected due to JIT compilations

**Comment by richvdh** [comment]
please see CONTRIBUTING.rst - this needs basing on `develop`, and needs sign-off.

**Comment by Valodim** [comment]
Well. The test runs work now, but the pypy-postgres run takes a whoopin' 43 minutes for some reason. Wonder what the problem there is... anyways, I'll leave this here for now :)

File: .travis.yml [hunk 1]
```diff
@@ -57,6 +57,19 @@ matrix:
     services:
       - postgresql
 
+  - python: 3.6
```
Comment (richvdh) [inline .travis.yml]: we're excluding python 3.6 deliberately: we get four workers on Travis, so this one just slows things down pointlessly.

**Comment by richvdh** [comment]
> the pypy-postgres run takes a whoopin' 43 minutes

Well, that sounds like a show-stopper :-p

File: .travis.yml (hunk 1 above)
Comment (Valodim) [reply .travis.yml]: ah, sorry. I think this one snuck during the rebase, didn't mean to add it

**Comment by Valodim** [comment]
Sure does. Even the regular test takes 2.5 times as long :(

There's three ways from here:
- figure out why it's slow and fix it :) heh. maybe if someone comes along who knows a lot about pypy
- keep only the pypy-sqlite test around. it's not prohibitively slow but should still catch most pypy breakage
- comment out the pypy tests in the CI pipeline, keeping pypy support in its current stepmotherly state. given that pypy's primary selling point is performance, but it's slower by a factor two in these (inadequate) tests, this is still a reasonable option

**Comment by luminoso** [comment]
@Valodim give it a try running the pypy tests with _"--jit off"_ to check if JIT compilations are the ones causing the tests taking so long

**Comment by luminoso** [comment]
Any news about this?

**Comment by richvdh** [comment]
seems to have stalled, so closing for now. We can reopen it if anyone wants to continue work.

**Pull request closed without merging**